### ISSUE TITLE
Provide access to k6 CLI definition

### ISSUE THREAD

**Issue by szkiba** [grafana/k6#4013]
### Feature Description

Creating k6 launchers (such as k6exec) would be greatly simplified by using k6 subcommand definitions. Without it, only workarounds can be created in certain use cases.

The root cause of the problem is that the launcher programs should parse the arguments of certain k6 subcommands. This is practically unsolvable without information about which flags exist and which ones have arguments. Duplication of definitions of k6 subcommands in launchers would cause continuous maintenance problems.

### Suggested Solution (optional)

Create a cobra.Command factory function for the k6 root command in the public k6 go API.

Something like this:

```go
package cmd

func New(ctx context.Context) (*cobra.Command, error) {
.
.
.
```

### Already existing or connected issues / PRs (optional)

The root cause of the following issues is the lack of k6 subcommand definitions:
- https://github.com/grafana/k6exec/issues/41
- https://github.com/grafana/k6exec/issues/48

**Comment by olegbespalov** [comment]
Personally, I see the value of the CLI definitions being exposed. However, it extends the k6 API surface, so probably good to hear the thoughts of the @grafana/k6-core. Do you have any concerns/thoughts about that?

**Comment by oleiade** [comment]
I was part of the initial discussions around this. Added it to the "nice to have" list of things we could include for v1 (or v2's) API surface. I'm rather strongly for it considering it would a positive impact on our ecosystem 🙇‍♂️

**Comment by joanlopez** [comment]
Yeah, I concur with both of you - I clearly see the tradeoff between slightly more maintenance work/compromise due to the extended API surface, and also see the benefits.

I'd suggest to evaluate such tradeoff, and if we consider it doesn't seem to put us in a risky position, I'd advocate to pushing for it 👍🏻
